server/src/main/kotlin/org/javacs/kt/definition/GoToDefinition.kt

```diff
@@ -87,4 +86,4 @@ fun goToDefinition(
     return destination
 }
 
-private fun isInsideJar(uri: String) = uri.contains(".jar!")
+private fun isInsideArchive(uri: String) = uri.contains("!")
```

Comment:
I am not sure if this is an edge case we've considered before, but can file names/path segments in URIs contain exclamation marks?

Reply:
Yeah, I suppose we never considered that before. URIs could potentially contain exclamation marks, but they could also contain stuff like `.jar!` before an actual jar, so I guess this was sort of a problem before already. I guess in those cases, we would need a more robust mechanism for this.

Reply:
I updated the check to be more strict (thus allowing paths with exclamation marks). There are still some edge cases I can think of that won't work properly, but these were already a problem before, so I think we can keep it like this for now.

I think we should probably try to refactor the whole class content provider (and KLS URI) logic at some point, since it's getting a bit too confusing IMO.